src/services/resource.js

```diff
@@ -210,6 +358,364 @@ module.exports = class resourceHelper {
+				if (inProgressResources.includes(item.id)) {
+					returnValue.reviewer_status = common.REVIEW_STATUS_INPROGRESS
+				}
+				returnValue.creator =
+					userDetails[item.user_id] && userDetails[item.user_id].name ? userDetails[item.user_id].name : ''
+				returnValue.organization = orgDetails[item.organization_id]
+				delete item.user_id
+				delete item.organization_id
+				return returnValue
+			})
+			result.count = response.count
+			result.in_progress_count = in_progress_count
+
+			return responses.successResponse({
+				statusCode: httpStatusCode.ok,
+				message: 'RESOURCE_LISTED_SUCCESSFULLY',
+				result,
+			})
+		} catch (error) {
+			throw error
+		}
+	}
+
+	static async findSequentialResources(organization_id, roles, resourceTypesInSequentialReview = []) {
+		// get unique user roles
+		const userRoleTitles = utils.getUniqueElements(roles.map((item) => item.title))
+
+		// fetch the resource wise review levels
+		const resourceWiseLevels = await this.fetchResourceReviewLevel(
+			organization_id,
+			userRoleTitles,
+			resourceTypesInSequentialReview
+		)
+		let innerArr = []
+		resourceTypesInSequentialReview.filter((item) => {
+			if (resourceWiseLevels[item]) {
+				innerArr.push({
+					[Op.and]: [{ type: item }, { next_stage: resourceWiseLevels[item] }],
+				})
+			}
+		})
+		const resourceFilter = {
+			[Op.or]: innerArr,
+			organization_id,
+			review_type: common.REVIEW_TYPE_SEQUENTIAL,
+			status: { [Op.in]: [common.RESOURCE_STATUS_SUBMITTED, common.RESOURCE_STATUS_IN_REVIEW] },
+		}
+		const resourcesDetails = await resourceQueries.resourceList(resourceFilter, ['id'])
+		const resoureId = resourcesDetails.result.map((item) => {
+			return item.id
+		})
+		return resoureId
+	}
+
+	static async findParallelResources(organization_id, resourceTypesInParallelReview = []) {
+		const resourceFilter = {
+			type: {
+				[Op.in]: resourceTypesInParallelReview,
+			},
+			organization_id,
+			review_type: common.REVIEW_TYPE_PARALLEL,
+			status: { [Op.in]: [common.RESOURCE_STATUS_SUBMITTED, common.RESOURCE_STATUS_IN_REVIEW] },
+		}
+		const resourcesDetails = await resourceQueries.resourceList(resourceFilter, ['id'])
+		const resoureId = resourcesDetails.result.map((item) => {
+			return item.id
+		})
+		return resoureId
+	}
+	// fetch all the resource ids from reviews table which is assigned to the reviewer
+	//  and already started reviewer
+	static async findResourcesAssignedToReviewer(organization_ids, reviewer_id) {
+		const reviewsFilter = {
+			reviewer_id,
+			organization_id: { [Op.in]: organization_ids },
+			status: { [Op.in]: common.PAGE_STATUS_VALUES.up_for_review },
+		}
+
+		const reviewsResponse = await reviewsQueries.findAll(reviewsFilter, ['resource_id'])
+		const res = reviewsResponse.map((item) => {
+			return item.resource_id
+		})
+		return res
+	}
+
+	static async findResourcesPickedUpByAnotherReviewer(reviewerId, openToAllResourcesMatchingMyLevel) {
```

Comment:
@adithyadinesh0412 reviewerId can rename to loggedInUserId for better understanding